### PR TITLE
Zipkin settings add thrift encoding

### PR DESCRIPTION
**Pull request by pavolloffay**
Related to #628

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@pavolloffay Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-sleuth&pullRequestId=650)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-sleuth&pullRequestId=650) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@pavolloffay Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-sleuth&pullRequestId=650)!

File: spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin/HttpZipkinSpanReporter.java [hunk 1]
```diff
@@ -28,7 +29,19 @@
...
+
+	/**
+	 * @param restTemplate {@link RestTemplate} used for sending requests to Zipkin
+	 * @param baseUrl       URL of the zipkin query server instance. Like: http://localhost:9411/
+	 * @param flushInterval in seconds. 0 implies spans are {@link #flush() flushed} externally.
+	 * @param spanMetricReporter service to count number of accepted / dropped spans
+	 * @param thrift use thrift encoding, default is JSON
+	 */
+	public HttpZipkinSpanReporter(RestTemplate restTemplate, String baseUrl, int flushInterval,
+								SpanMetricReporter spanMetricReporter, boolean thrift) {
```
Comment (adriancole) [inline spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin/HttpZipkinSpanReporter.java]: there's another json encoding coming soon.. not sure we'll want to use boolean here, especially as a property. Probably a stringification of the enum name is a better choice

File: spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin/ZipkinProperties.java [hunk 2]
```diff
@@ -33,6 +33,7 @@
 	private String baseUrl = "http://localhost:9411/";
 	private boolean enabled = true;
 	private int flushInterval = 1;
+	private boolean thrift = false;
```
Comment (adriancole) [inline spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin/ZipkinProperties.java]: ex `private Encoding encoding = Encoding.JSON;` assuming property-fying works as it should

**Comment by pavolloffay** [comment]
@adriancole thanks, PR updated

File: spring-cloud-sleuth-zipkin/src/test/java/org/springframework/cloud/sleuth/zipkin/HttpZipkinSpanReporterTest.java [hunk 3]
```diff
@@ -15,17 +26,11 @@
...
+
 import zipkin.Span;
 import zipkin.junit.HttpFailure;
 import zipkin.junit.ZipkinRule;
-
-import java.util.ArrayList;
-import java.util.Random;
-import java.util.concurrent.atomic.AtomicReference;
-
-import static java.util.Arrays.asList;
```
Comment (adriancole) [inline spring-cloud-sleuth-zipkin/src/test/java/org/springframework/cloud/sleuth/zipkin/HttpZipkinSpanReporterTest.java]: sorry to ask, but pls undo the reordering here

**Comment by adriancole** [comment]
@marcingrzejszczak is there any bookkeeping to do for new properties? (ex some doc needs updating)

**Comment by marcingrzejszczak** [comment]
I'll add the javadocs

**Comment by adriancole** [comment]
thx @pavolloffay!

**Pull request merged**